website/scripts/index_search_content.js

```diff
@@ -1,3 +1,99 @@
-const { indexDocsContent } = require('@hashicorp/react-search/tools')
+require('dotenv').config()
+const fs = require('fs')
+const path = require('path')
+const {
+  indexContent,
+  getDocsSearchObject,
+} = require('@hashicorp/react-search/tools')
+const resolveNavData = require('../components/remote-plugin-docs/utils/resolve-nav-data')
+const fetchGithubFile = require('../components/remote-plugin-docs/utils/fetch-github-file')
 
-indexDocsContent()
+const GITHUB_API_TOKEN = process.env.CI_GITHUB_TOKEN
+
+// Run indexing
+indexContent({ getSearchObjects })
+
+async function getSearchObjects() {
+  // Set up an array to collect all search objects
+  const searchObjects = []
+  // Resolve /docs, /guides, and /intro nav data, which
+  // corresponds to all the content we will actually render
+  // This avoids indexing non-rendered content, and partials.
+  // `docs` content
+  const docsNav = await resolveNavData(
+    'data/docs-nav-data.json',
+    'content/docs',
+    {
+      remotePluginsFile: 'data/docs-remote-plugins.json',
+      githubToken: GITHUB_API_TOKEN,
+    }
+  )
+  const docsObjects = await searchObjectsFromNavData(docsNav, 'docs')
+  searchObjects.push(...docsObjects)
+  // `guides` content
+  const guidesNav = await resolveNavData(
+    'data/guides-nav-data.json',
+    'content/guides'
+  )
+  const guidesObjects = await searchObjectsFromNavData(guidesNav, 'guides')
+  searchObjects.push(...guidesObjects)
+  // `intro` content
+  const introNav = await resolveNavData(
+    'data/intro-nav-data.json',
+    'content/intro'
+  )
+  const introObjects = await searchObjectsFromNavData(introNav, 'intro')
+  searchObjects.push(...introObjects)
+  // Return the collected search objects
+  return searchObjects
```

Comment:
Nit: we might want to consider executing these in parallel, instead of serially. Should be able to do an `await Promise.all([...])` and have something like this for each section:
```js
(async () => {
  const docsNav = await resolveNavData(
    'data/docs-nav-data.json',
    'content/docs',
    {
      remotePluginsFile: 'data/docs-remote-plugins.json',
      githubToken: GITHUB_API_TOKEN,
    }
  )
  const docsObjects = await searchObjectsFromNavData(docsNav, 'docs')
  searchObjects.push(...docsObjects)
})()
```

Reply:
Yeah, that's probably better 😄

Reply:
Oooh great point! ✨ I tried this out and noticed a _really_ slight difference in execution time:

![CleanShot 2021-03-09 at 18 13 47@2x](https://user-images.githubusercontent.com/4624598/110551455-88816b00-8103-11eb-8a3a-502e7bdb46ba.png)

Personally I think it's a little harder to parse what's going on with the `Promise.all` - so IMO the performance boost doesn't feel worth it. That being said this definitely comes down to opinion, may just be that I'm less used to seeing `Promise.all`, so happy to change if it doesn't feel like there's a difference in readability for y'all! 🙇

Reply:
Agreed, I'm fine to leave it in that case! Nice due-diligence ✨

Reply:
okay, updated! Had to flatten with a `reduce` after `await Promise.all([fetchDocsObjects(), /* ...etc */])`, but still feels pretty clean.